### ISSUE TITLE
Lack of code formatting assistance

### ISSUE THREAD

**Issue by Tyrrrz** [AmmyUI/AmmyUI#112]
It's quite hard to maintain good formatting because automatic indentation is not applied on copy/pasting and in some cases when making a new line. Also, Visual Studio defaults to 4 spaces per tab instead of 2 and Ammy doesn't have its own page in Settings > Text Editor (it seems to use configuration from Plain Text editor, which is not very convenient).

**Comment by ionoy** [comment]
There is an auto-formatter in the settings page (top-right corner).

**Comment by Tyrrrz** [comment]
That's true, however it also removes empty newlines which I often use to separate parts of markup. Plus having to click on it by yourself is a little inconvenient, especially since Visual Studio takes care of indentation in every other part of the code.

**Comment by ionoy** [comment]
I'm currently swamped by LiveXAML development, so it's hard for me to find much time for Ammy. You could help if you want.

Formatting code is here: https://github.com/AmmyUI/AmmyUI/blob/master/src/IDE/Ammy.VisualStudio.Service/Intellisense/CustomFormatter.cs

I could help you with advice since I'm usually online.

**Comment by Tyrrrz** [comment]
Yes, I will definitely see if I can make a pull request for any of the issues I had. For now I just created the issues because I don't have too much time myself either. :p
